src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-form/osd-form.component.ts

```diff
@@ -209,16 +246,50 @@ export class OsdFormComponent extends CdForm implements OnInit {
     }
   }
 
+  emitDeploymentSelection() {
+    const options = document.querySelector(
+      'input[name="deploymentOptions"]:checked'
+    ) as HTMLInputElement;
+    this.emitDeploymentOption.emit(options?.value);
+  }
```

Comment:
That's not very Angular-ish. As a rule of thumb, [we shouldn't use query selectors and directo DOM manipulation/inspection](https://angular.io/api/core/ElementRef#properties)... There [are better/more Angular-ish ways](https://indepth.dev/posts/1336/how-to-do-dom-manipulation-properly-in-angular):
* [`ViewChild`/`ViewChildren`](https://angular.io/api/core/ViewChildren),
* Or given we're using (supposed to) the Reactive Forms, [this specific approach](https://stackoverflow.com/questions/43713558/how-to-get-a-single-value-from-formgroup).

Reply:
Updated it and used the reactive form one